js/shared.js

```diff
@@ -374,3 +376,160 @@ async function findMatchingIncipits(incipit, titleStart) {
+            method: "POST",
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json;charset=utf-8"
+            },
+            body: JSON.stringify(referencesAnnosQuery)
+        })
+        .then(response => response.json())
+        .then(annos => {
+            if(!witnessesObj.hasOwnProperty(witnessURI)) witnessesObj[witnessURI] = {}
+            witnessesObj[witnessURI].glosses = new Set([...glossUriSet, ...new Set(annos.map(anno => anno.body.references.value).flat())])
+            glossUriSet = new Set([...glossUriSet, ...new Set(annos.map(anno => anno.body.references.value).flat())])
+            return Promise.resolve(witnessesObj)
+        })
+        .catch(err => {
+            console.error(err)
+            return Promise.reject([])
+        }))
+
+        all.push(fetch(`${__constants.tiny}/query?limit=100&skip=0`, {
+            method: "POST",
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json;charset=utf-8"
+            },
+            body: JSON.stringify(selectionsAnnosQuery)
+        })
+        .then(response => response.json())
+        .then(annos => {
+            if(!witnessesObj.hasOwnProperty(witnessURI)) witnessesObj[witnessURI] = {}
+            const existingSelections = witnessesObj[witnessURI].selections ? witnessesObj[witnessURI].selections : []
+            witnessesObj[witnessURI].selections = [...existingSelections, ...annos.map(anno => anno.body.selections.value).flat()]
+            return Promise.resolve(witnessesObj)
+        })
+        .catch(err => {
+            console.error(err)
+            return Promise.reject([])
+        })
+        )
+    }
+
+    // This has the asyncronous behavior necessary to build witnessesObj.
+    Promise.all(all)
+    .then(success => {
+        witnessesObj.referencedGlosses = glossUriSet
+        for(const witnessURI in witnessesObj){
+            if(witnessURI === "referencedGlosses") continue
+            const witnessInfo = witnessesObj[witnessURI]
+            witnessInfo.glosses.forEach(glossURI => {
+                // For each Gloss URI find its corresponding 'attach' button and ensure it is classed as a Gloss that is already attached to this source.
+                document.querySelectorAll(`.toggleInclusion[data-id="${glossURI}"]`).forEach(btn => {
+                    btn.classList.add("attached-to-source")
+                })    
+            })
+            preselectLines(witnessInfo.selections, witnessForm, false)
+        }
+    })
+    .catch(err => {
+        console.error("Witnesses Object Error")
+        console.error(err)
+        const ev = new CustomEvent("Witnesses Object Error")
+        globalFeedbackBlip(ev, `Witnesses Object Error`, false)
+    })
+    
+}
+
+/**
+ * Undo a manual user selection of text on the page.  Mainly just a UI trick.
+ * 
+ * @param s The Browser Selection object
+ */ 
+function undoBrowserSelection(s){
+    if (s) {
```

Comment:
This is complicated. Is this the same/better?

`s?.removeAllRanges?.() ?? s?.empty?.()`

I get that it looks goofy, but it is still readable. I'd really like to dump the else if at least.